src/vcpkg/base/system.cpp

```diff
@@ -292,7 +292,8 @@ namespace vcpkg
 
     const ExpectedS<Path>& get_system32() noexcept
     {
-        static const ExpectedS<Path> s_system32 = get_system_root().map([](const Path& p) { return p / "System32"; });
+        // This needs to be lowercase or msys-ish tools break. See https://github.com/microsoft/vcpkg-tool/pull/418/
+        static const ExpectedS<Path> s_system32 = get_system_root().map([](const Path& p) { return p / "system32"; });
```

Comment:
>Shouldn't we expect the name system32 and search in the windows folder to confirm it?

No, because the name in the Windows folder is always "System32" (with a capital) which is what breaks msys.